### PR TITLE
split file into modules and combine all typing declaration

### PR DESCRIPTION
**Pull request by sirahd**
- split file into `parseOption.ts`, `formatFieldValue.ts`, `supplementField.ts` and `supplementField.ts` based on semantics.
- combine all the typings declaration into one `options.ts`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,8 @@
   },
   "browserify-shim": {
     "vega": "global:vega",
-    "vega-lite": "global:vl"
+    "vega-lite": "global:vl",
+    "d3-format": "global:d3",
```
Comment (domoritz) [inline package.json]: Let's include the d3 libraries in the build. It should not be that large.

**Comment by sirahd** [comment]
@domoritz Done!

File: index.html [hunk 2]
```diff
@@ -4,10 +4,16 @@
   <meta charset="utf-8">
 
   <!-- Tooltip Library -->
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/4.0.0/d3.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/vega/3.0.0-beta.31/vega.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/vega-lite/2.0.0-beta.2/vega-lite.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/vega-embed/3.0.0-beta.14/vega-embed.min.js" charset="utf-8"></script>
+  <script src="https://d3js.org/d3-format.v1.min.js"></script>
```
Comment (domoritz) [inline index.html]: should not be necessary if we don't shim d3 libs

**Comment by domoritz** [comment]
How big is the library now? Can you create a test file here you only use Vega to make sure it works?

**Comment by sirahd** [comment]
The original file is ~300KB and minified one is 43KB. I also separated the examples into vega and vega-lite. And yes, vega-tooltip works without having to import vega-lite

File: vega-lite-examples.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  <!-- Example Page -->
+  <script type="text/javascript" src="vega-lite-examples.js"></script>
+  <link rel="stylesheet" type="text/css" href="examples.css">
+
+</head>
+
+<body>
+  <h1>Vega-Lite Tooltip Examples</h1>
+
```
Comment (domoritz) [inline vega-lite-examples.html]: Please add a prominent link back to the main page.

File: src/supplementField.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      }
+    }
+  }
+
+  // return undefined if no match was found
+  return undefined;
+}
+
+/**
+* Supplement a fieldOption (from options.fields[]) with a fieldDef, config
```
Comment (domoritz) [inline src/supplementField.ts]: I think this should be indented by 1.

**Comment by kanitw** [comment]
43 KB is still worth splitting IMHO.

**Comment by domoritz** [comment]
Looks good to me. I'll merge once you fix the two minor comments I have.

**Comment by kanitw** [comment]
(but splitting can be done in a separate PR)

**Comment by domoritz** [comment]
> 43 KB is still worth splitting IMHO.

Let's look at how we can reduce the size further first.

**Comment by sirahd** [comment]
I think we can further reduce the size by removing d3 map. I'll do that after we merge this PR

**Comment by domoritz** [comment]
Use https://github.com/hughsk/disc and see what is actually big.

File: src/supplementField.ts (hunk 4 above)
Comment (sirahd) [inline src/supplementField.ts]: Do you mean the comment block?

**Comment by sirahd** [comment]
@domoritz Done!

**Pull request merged**